### DISCUSSION TITLE
Welcome to rocky-tools Discussions!

### DISCUSSION THREAD

**Discussion by pajamian** [pajamian/rocky-tools#1] [General]
Discussions for migrate2rocky

**Comment by scotsgeezer** [comment]
Started with CentOS8 build 1905, not updated

Run the migratetorocky script from https://github.com/pajamian/rocky-tools/blob/main/migrate2rocky.sh

This failed miserably.

Updated to CentOS8.3 with dnf update

Rerun the above script, took some time as it reinstalled rocky and the repos etc....

Reboot the machine, which is hosted on vSphere 6.7U3

Got the efi secure boot screen

I shut the machine and edited the VM config file edit>VMoptions>boot options>Enable UEFI secure boot  ----> unchecked ie. switched off secure boot

Powered on, went straight to grub prompt

set root=(lvm/cl-root)
linux (hd0,gpt2)/vmlinuz-4.18.0-240.22.1.el8_3.x86_64 root=/dev/cl/root
initrd (hd0,gpt2)/initramfs-4.18.0-240.22.1.el8_3.x86_64.img
boot

The machine booted into Rocky

For the system to reboot correctly I had to run:
grub2-mkconfig -o /boot/efi/EFI/rocky/grub.cfg

Secure boot still disabled until Rocky gets enabled for secure boot ( I presume)

**Reply by electroniceel** [reply]
I just tested migrate2rocky.sh on an UEFI machine (vm running on qemu/kvm) with current CentOS 8.3.2011:

The conversion worked nicely, but I had to run
grub2-mkconfig -o /boot/efi/EFI/rocky/grub.cfg

afterwards to make it boot.

But what was missing is making rocky known to the uefi firmware. I suggest to run something like this for it:
efibootmgr -c -d /dev/vda -L "Rocky Linux" -l /EFI/rocky/grubx64.efi

The correct disk for the -d parameter needs to be detected of course. I suggest something like
findmnt --mountpoint /boot/efi --output SOURCE --noheadings
to get it.

What I was also seeing is that the grub menu still shows centos entries afterwards. This is because the /boot/loader/entries/ dir still contains entries for Centos and the kernel-rpms from Centos are still installed. Once the centos kernel-rpms are uninstalled (or get uninstalled later), the entries will be removed automatically.

There is only one issue though: the rescue kernel entry. That is not owned by a rpm and will thus never be removed. So as it is now, it will always be centos and will always show up in grub as such. So maybe create a new rescue kernel entry? Or ignore it?

**Reply by scotsgeezer** [reply]
Good info EEel.

I tried this:
[root@c82rocky ~]# ls -al /boot/loader/entries/
total 24
drwx------. 2 root root 4096 May 19 14:55 .
drwxr-xr-x. 3 root root 4096 May 18 15:24 ..
-rw-r--r--. 1 root root  395 May 18 15:26 42fb7612fda744f58da985f2bdc2234b-0-rescue.conf
-rw-r--r--. 1 root root  351 May 18 17:19 42fb7612fda744f58da985f2bdc2234b-4.18.0-240.22.1.el8_3.x86_64.conf.orig <-----*** renamed
-rw-r--r--. 1 root root  339 May 19 10:48 42fb7612fda744f58da985f2bdc2234b-4.18.0-240.22.1.el8.x86_64.conf
-rw-r--r--. 1 root root  318 May 18 15:26 42fb7612fda744f58da985f2bdc2234b-4.18.0-80.el8.x86_64.conf

After the rename i reran grub2-mkconfig -o /boot/efi/EFI/rocky/grub.cfg

This removed the offending item.

**Reply by electroniceel** [reply]
I don't think patching/renaming these entries is a good idea. Because these files are owned by a still-installed kernel rpm. You can check with rpm -qf <full-filename>.

So the solution there would be to uninstall the centos kernel rpm. But not sure if this is a good idea, because for the time being, the migration from centos to rocky will for most people also be a migration from 8.3 to 8.4 at the same time (centos will probably take ages until they release their 8.4). That makes problems more likely, e.g. due to driver issues or similar. Then it would be a good idea to leave the user the rpm from the kernel they were previously running.

Only the -rescue is not owned by a kernel rpm. But patching the description, while still using the centos kernel seems odd to me. So if we want to get rid of it, it has to be replaced by a rocky rescue kernel.

**Reply by pajamian** [reply]
The reason we can't uninstall the current CentOS kernel is because it's a running kernel at the time of the upgrade and it's a generally bad idea to uninstall a running kernel (dnf explicitly prevents this unless you jump through hoops to override it).  migrate2rocky does, however, install the newest RockyLinux kernel and should set it as default.
As an aside, I would be curious if someone wants to test the migration on a box with an elrepo kernel.

**Comment by mikaku** [comment]
Here are my thoughts:

- The messages in color blue are so dark that I can't read them (consider using a different color).
- The log generated is named as `centos2rocky.log` instead of `migrate2rocky.log` (which is the name of the script who has generated it).
- The ANSI color sequences appear saved in the log. I think you could omit the colors when the output file is not a tty.

Besides this, the update went smooth and completely successful. :+1: 

PS: Colors are fine. I would just play with green and red colors to indicate if the process went good or not.

**Reply by pajamian** [reply]
Since this isn't the first complaint about colors I might just get rid of them.  The issue with logging colors is that the log is simply tee'd off of stdout and stderr, so if colors go to the console they go to the log, changing that would require a significant rewrite (although not impossible).
